.github/workflows/reusable-lint.yaml

```diff
@@ -1,210 +1,217 @@
-name: Run linters
+# Lint Check Workflow
+#
+# This workflow runs various linting tools on the repository to ensure code quality
+# and adherence to coding standards. It supports configurable linting options
+# through GitHub variables and can be triggered on pull requests or called manually.
+#
+# Supported linters:
+# - Black (code formatting)
+# - Doc8 (documentation formatting)
+# - Flake8 (code style and quality)
+# - Bandit (security vulnerability scanning)
+#
+# The workflow dynamically enables/disables linters based on repository configuration.
+
+name: Run linters  # Workflow name
+
+# Set default permissions for all jobs to read-only
+permissions:
+  contents: read
 
 on:
-  # Reusable workflow
-  workflow_call:
+  pull_request:  # Trigger on pull request events
```

Comment:
Again,  I think this is superfluous here,  because the triggers will have to be in the "ruleset" work flows.  Though I don't think it really matters (would just potentially cause duplicate runs in geoips ci repo)